src/Symfony/Component/Validator/Constraints/ExpressionLanguageSyntax.php

```diff
@@ -31,14 +31,20 @@ class ExpressionLanguageSyntax extends Constraint
     public $message = 'This value should be a valid expression.';
     public $service;
     public $allowedVariables;
+    public $allowNullAndEmptyString = false;
 
-    public function __construct(array $options = null, string $message = null, string $service = null, array $allowedVariables = null, array $groups = null, $payload = null)
+    public function __construct(array $options = null, string $message = null, string $service = null, array $allowedVariables = null, bool $allowNullAndEmptyString = null, array $groups = null, $payload = null)
     {
         parent::__construct($options, $groups, $payload);
 
         $this->message = $message ?? $this->message;
         $this->service = $service ?? $this->service;
         $this->allowedVariables = $allowedVariables ?? $this->allowedVariables;
+        $this->allowNullAndEmptyString = $allowNullAndEmptyString ?? $this->allowNullAndEmptyString;
+
+        if (isset($options['allowNullAndEmptyString'])) {
+            trigger_deprecation('symfony/validator', '5.4', sprintf('The "allowNullAndEmptyString" option of the "%s" constraint is deprecated.', self::class));
+        }
```

Comment:
This parameter is our forward compatibility layer and it allows us to opt-in to the behavior we want to have in 6.0. An application that does not explicitly enable that new behavior will potentially break when upgrading to Symfony 6.

What we should deprecate is not explicitly setting it to `true`.

```suggestion
        if (!$this->allowNullAndEmptyString) {
            trigger_deprecation('symfony/validator', '5.4', 'Not setting "allowNullAndEmptyString" of the "%s" constraint to "true" is deprecated.', __CLASS__);
        }
```